cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -123,15 +123,14 @@ func isDeviceModelUpdated(oldTwin *v1alpha2.DeviceModel, newTwin *v1alpha2.Devic
 // deviceModelUpdated is function to process updated deviceModel
 func (dc *DownstreamController) deviceModelUpdated(deviceModel *v1alpha2.DeviceModel) {
 	value, ok := dc.deviceModelManager.DeviceModel.Load(deviceModel.Name)
-	dc.deviceModelManager.DeviceModel.Store(deviceModel.Name, deviceModel)
 	if ok {
 		cachedDeviceModel := value.(*v1alpha2.DeviceModel)
 		if isDeviceModelUpdated(cachedDeviceModel, deviceModel) {
 			dc.updateAllConfigMaps(deviceModel)
 		}
-	} else {
-		dc.deviceModelAdded(deviceModel)
 	}
+
+	dc.deviceModelAdded(deviceModel)
```

Comment:
I think this would be better if it were to be changed
```go
func (dc *DownstreamController) deviceModelUpdated(deviceModel *v1alpha2.DeviceModel) {
        value, ok := dc.deviceModelManager.DeviceModel.Load(deviceModel.Name)
        if !ok {
                dc.deviceModelAdded(deviceModel)
                return
        }

        dc.deviceModelManager.DeviceModel.Store(deviceModel.Name, deviceModel)

        cachedDeviceModel := value.(*v1alpha2.DeviceModel)
        if isDeviceModelUpdated(cachedDeviceModel, deviceModel) {
                dc.updateAllConfigMaps(deviceModel)
        }
}
```
The `deviceModelAdded` may have some other operations in the future and it may not be a good idea to always call it directly in the `deviceModelUpdated`